### PR TITLE
Fix nexus upgrade problem

### PR DESCRIPTION
**Pull request by talebipour**
+ Create specific database for tags
+ Replace docker-compose-wrapper with testcontainers
+ Add UpgradeTest
+ Change test workflow to test plugin with 3.28.0 version

### CODE REVIEW

File: src/main/java/ir/sahab/nexus/plugin/tag/internal/TagStore.java [hunk 1]
```diff
@@ -100,7 +141,7 @@ public Tag addOrUpdate(TagDefinition definition) {
                 log.info("Tag {} updated in database.", entity);
             } else {
                 entityAdapter.addEntity(tx, entity);
-                log.info("Tag {} added to database.", entity);
+                log.info("Tag {} added to database.!!!", entity);
```
Comment (MSDehghan) [inline src/main/java/ir/sahab/nexus/plugin/tag/internal/TagStore.java]: Why "!!!"?

File: src/test/java/ir/sahab/nexus/plugin/tag/NexusContainerUtil.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                .withClasspathResourceMapping("/nexus.properties", "/nexus-data/etc/nexus.properties",
+                        BindMode.READ_ONLY)
+                .withExposedPorts(PORT)
+                .waitingFor(new HttpWaitStrategy().forPort(PORT).withStartupTimeout(Duration.ofMinutes(10)))
+                .withLogConsumer(outputFrame -> System.err.println(outputFrame.getUtf8String()));
+
+        if (extraBinds.length > 0) {
+            container = container.withCreateContainerCmdModifier(cmd -> {
+                Bind[] existingBinds = cmd.getHostConfig().getBinds();
+                Bind[] binds = Arrays.copyOf(existingBinds, existingBinds.length + existingBinds.length);
```
Comment (MSDehghan) [inline src/test/java/ir/sahab/nexus/plugin/tag/NexusContainerUtil.java]: i think it should be like this:
```suggestion
                Bind[] binds = Arrays.copyOf(existingBinds, existingBinds.length + extra.length);
```

File: src/main/java/ir/sahab/nexus/plugin/tag/internal/TagStore.java (hunk 1 above)
Comment (talebipour) [reply src/main/java/ir/sahab/nexus/plugin/tag/internal/TagStore.java]: It was for test, I forgot to remove it.

File: src/test/java/ir/sahab/nexus/plugin/tag/NexusContainerUtil.java (hunk 2 above)
Comment (talebipour) [reply src/test/java/ir/sahab/nexus/plugin/tag/NexusContainerUtil.java]: Done

**Review by MSDehghan — APPROVED** [review]

**Pull request merged**